stdlib/Sockets/src/Sockets.jl

```diff
@@ -806,6 +806,7 @@ socket is connected to. Valid only for connected TCP sockets.
 getpeername(sock::TCPSocket) = _sockname(sock, false)
 
 function _sockname(sock, self=true)
+    check_open(sock)
```

Comment:
It is also valid to use this function on a socket which is not open, but is initialized:
```suggestion
    sock.status == StatusInit || check_open(sock)
```